src/python/impactx/dashboard/Input/generalFunctions.py

```diff
@@ -0,0 +1,273 @@
+        :param input_value: The value to validate.
+        :param value_type: The desired type ('int', 'float', 'str').
+        :return: A list of error messages. An empty list if there are no errors.
+        """
+
+        if value_type == "int":
+            if input_value is None:
+                return ["Must be an integer"]
+            try:
+                value = int(input_value)
+                if value < -10000000000:
+                    return ["Must be positive"]
+                return []
+            except ValueError:
+                return ["Must be an integer"]
+
+        elif value_type == "float":
+            if input_value is None:
+                return ["Must be a float"]
+            try:
+                value = float(input_value)
+                if value < -1000000000:
+                    return ["Must be positive"]
+                return []
+            except ValueError:
+                return ["Must be a float"]
+
+        elif value_type == "str":
+            if input_value is None:
+                return ["Must be a string"]
+            return []
+
+        else:
+            return ["Unknown type"]
+
+    @staticmethod
+    def update_simulation_validation_status():
+        """
+        Checks if any input fields are not provided with the correct input type.
+        Updates the state to enable or disable the run simulation button.
+        """
+
+        error_details = []
+
+        # Check for errors in distribution parameters
+        for param in state.selectedDistributionParameters:
+            if param["parameter_error_message"]:
+                error_details.append(
+                    f"{param['parameter_name']}: {param['parameter_error_message']}"
+                )
+
+        # Check for errors in lattice parameters
+        for lattice in state.selectedLatticeList:
+            for param in lattice["parameters"]:
+                if param["parameter_error_message"]:
+                    error_details.append(
+                        f"Lattice {lattice['name']} - {param['parameter_name']}: {param['parameter_error_message']}"
+                    )
+
+        # Check for errors in input card
+        if state.npart_validation:
+            error_details.append(f"Number of Particles: {state.npart_validation}")
+        if state.kin_energy_validation:
+            error_details.append(f"Kinetic Energy: {state.kin_energy_validation}")
+        if state.bunch_charge_C_validation:
+            error_details.append(f"Bunch Charge: {state.bunch_charge_C_validation}")
+        if state.selectedLatticeList == []:
+            error_details.append("LatticeListIsEmpty")
+
+        # Print all collected error messages
+        """
+        if error_details:
+            print("Errors detected in the following parameters:")
+            for error in error_details:
+                print(ANSI_RED + error + ANSI_RESET)
+        """
```

Comment:
Does this print or is this a multi-line string that does nothing?

Reply:
Debug code. Can remove.